be/src/connector/jdbc_connector.cpp

```diff
@@ -49,7 +49,13 @@ static std::string get_jdbc_sql(const Slice jdbc_url, const std::string& table,
         }
     }
     if (limit != -1) {
-        oss << " LIMIT " << limit;
+        if (jdbc_url.starts_with("jdbc:oracle")) {
+            // oracle doesn't support limit clause, we should generate a subquery to do this
+            // ref: https://stackoverflow.com/questions/470542/how-do-i-limit-the-number-of-rows-returned-by-an-oracle-query-after-ordering
+            return fmt::format("SELECT * FROM ({}) WHERE ROWNUM <= {}", oss.str(), limit);
```

Comment:
Are you sure we don't have to add an alias for subquery?

Reply:
> Are you sure we don't have to add an alias for subquery?

I have tested it and it seems to be OK